src/tools/rados/rados.cc

```diff
@@ -1092,14 +1107,36 @@ class RadosBencher : public ObjBencher {
 
   int aio_read(const std::string& oid, int slot, bufferlist *pbl, size_t len,
 	       size_t offset) override {
-    return io_ctx.aio_read(oid, completions[slot], pbl, len, offset);
+    int ret = 0;
+    if (destination & OP_DEST_OBJ) {
+      ret = io_ctx.aio_read(oid, completions[slot], pbl, len, offset);
```

Comment:
If no objections, I would like to unify the data read with OMAP / attrs through the `ObjectReadOperation`.